### ISSUE TITLE
HTML colour names shouldn't always be compressed

### ISSUE THREAD

**Issue by leebenson** [sass/sass#1803]
HTML colour names are incorrectly compressed in a non-colour context.

This breaks any accompanying HTML code that use attribute strings that may happen to clash with HTML colour names.

Example Sass:

```
button[data-fillcolour='white'] {
  background: white;
}
```

Expected after compression (expanded for readability):

```
button[data-fillcolour=white] {
  background: #fff;
}
```

Actual:

```
button[data-fillcolour=#fff] {
  background: #fff;
}
```

**Comment by chriseppstein** [comment]
I can't reproduce this with the latest release of Sass. Are you sure this exact code causes the issue you described?

The only way I can reproduce it is to have source like this:

``` scss
$fill-color: white;
button[data-fillcolour='#{$fill-color}'] {
  background: $fill-color;
}
```

Which is wrong. because color names are always colors in a SassScript value context. Instead, use unquoted strings if you need to ensure a particular output format for a color.

**Comment by leebenson** [comment]
Actually, I was originally using unquoted strings, and got an error from the compiler that strings couldn't be unquoted. I added the quotes and got the above error. When I switched to cssnano instead of sass' compression, the problem disappeared.

It's possible I just goofed the original syntax somehow to generate the error. I haven't been able to replicate it since. Thanks for looking at it anyway.
